src/app/filters/filter-fields.component.html

```diff
@@ -34,11 +35,38 @@
           <li *ngFor="let query of currentField.queries"
               [ngClass]="{'selected': query.value === currentValue}">
             <a role="menuitem" tabindex="-1" (click)="selectValue(query)">
+              <span *ngIf="query?.iconClass" class="fa {{query?.iconClass}}"></span>
+              <img class="avatar" *ngIf="query?.imageUrl" [attr.src]="query?.imageUrl" />
               {{query.value}}
             </a>
           </li>
         </ul>
       </div>
     </div>
+    <div *ngIf="currentField.type === 'typeahead'">
+      <div class="btn-group bootstrap-select form-control filter-select" dropdown>
+        <div class="pull-left typeahead-input-container" (click)="emitSelectedField()" dropdownToggle>
+          <input class="form-control" type="{{currentField.type}}" [(ngModel)]="currentValue"
+            placeholder="{{currentField.placeholder}}" (keypress)="onValueKeyPress($event)"
+            (keyup)="filterList($event)" #searchField>
+          <span class="caret"></span>
+        </div>
+          <ul class="dropdown-menu" role="menu" #listData dropdownMenu>
+            <li *ngIf="currentField.placeholder">
+              <a role="menuitem" tabindex="-1" (click)="selectValue()">
+                {{currentField.placeholder}}
+              </a>
+            </li>
+            <li *ngFor="let query of filteredList"
+              [ngClass]="{'selected': query.value === currentValue}">
+              <a role="menuitem" tabindex="-1" (click)="selectValue(query)">
+                <span *ngIf="query?.iconClass" class="fa {{query?.iconClass}}"></span>
+                <img class="avatar" *ngIf="query?.imageUrl" [attr.src]="query?.imageUrl" />
+                  <span [innerHTML]="query.value | almSearchHighlight: searchField.value"></span>
```

Comment:
That's definitely a possibility. Although, I was thinking that the highlighting for this menu should be the same as the existing 'select' dropdown menu. Otherwise, it's probably something to run by UXD.